test/management/hooks.tests.js

```diff
@@ -0,0 +1,143 @@
+var expect = require('chai').expect;
+var nock = require('nock');
+
+var SRC_DIR = '../../src';
+var API_URL = 'https://tenant.auth0.com';
+
+var HooksManager = require(SRC_DIR + '/management/HooksManager');
+var ArgumentError = require('rest-facade').ArgumentError;
+
+const hook_template = `/**
+@param {object} client - information about the client
+@param {string} client.name - name of client
+@param {string} client.id - client id
+@param {string} client.tenant - Auth0 tenant name
+@param {object} client.metadata - client metadata
+@param {array|undefined} scope - array of strings representing the scope claim or undefined
+@param {string} audience - token's audience claim
+@param {object} context - additional authorization context
+@param {object} context.webtask - webtask context
+@param {function} cb - function (error, accessTokenClaims)
+*/
+module.exports = function(client, scope, audience, context, cb) {
+  var access_token = {};
+  access_token.scope = scope;
+
+  // Modify scopes or add extra claims
+  // access_token['https://example.com/claim'] = 'bar';
+  // access_token.scope.push('extra');
+
+  cb(null, access_token);
+};
+`;
+
+describe('HooksManager', function() {
+  before(function() {
+    this.token = 'TOKEN';
+    this.hooks = new HooksManager({
+      headers: { authorization: 'Bearer ' + this.token },
+      baseUrl: API_URL
+    });
+  });
+
+  describe('instance', function() {
+    var methods = ['get'];
+
+    methods.forEach(function(method) {
+      it('should have a ' + method + ' method', function() {
+        expect(this.hooks[method]).to.exist.to.be.an.instanceOf(Function);
+      });
+    });
+  });
+
+  describe('#constructor', function() {
+    it('should error when no options are provided', function() {
+      expect(HooksManager).to.throw(ArgumentError, 'Must provide manager options');
+    });
+
+    it('should throw an error when no base URL is provided', function() {
+      var client = HooksManager.bind(null, {});
+
+      expect(client).to.throw(ArgumentError, 'Must provide a base URL for the API');
+    });
+
+    it('should throw an error when the base URL is invalid', function() {
+      var client = HooksManager.bind(null, { baseUrl: '' });
+
+      expect(client).to.throw(ArgumentError, 'The provided base URL is invalid');
+    });
+  });
+
+  describe('#get', function() {
+    beforeEach(function() {
+      this.data = {
+        triggerId: 'credentials-exchange',
+        name: 'cce-hook',
+        active: true,
+        code: hook_template,
+        secrets: {
+          'api-key': 'my custom api key'
+        },
+        dependencies: {
+          bcrypt: '3.0.6'
+        }
+      };
+
+      this.request = nock(API_URL)
+        .get('/hooks/' + this.data.name)
```

Comment:
while this would be something different than `undefined`, the definition of the API says it should be the `id` property of the hook. Remember tests are also documentation.